### ISSUE TITLE
Expose AuthContext

### ISSUE THREAD

**Issue by mykeels** [PropelAuth/react#6]
## Summary

It could be helpful to have a way to wrap components with the `AuthContext.Provider` when using storybook.

## The Problem

I want to be able to test components that use `@propelauth/react`, in isolation, without needing to rely on an an auth API endpoint and its data.

## The Proposed Solution

I found that by passing:

```tsx
<AuthContext.Provider
      value={{
        authInfo: {
          user: {
            email: 'john.doe@example.com',
            emailConfirmed: true,
            enabled: true,
            locked: false,
            mfaEnabled: false,
            userId: 'john.doe',
            username: 'John Doe',
            firstName: 'John',
            lastName: 'Doe',
          },
          accessToken:
            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c',
          expiresAtSeconds: 300,
          orgHelper: {
            getOrg: () => undefined,
            getOrgByName: () => undefined,
            getOrgIds: () => [],
            getOrgs: () => [],
          },
        },
        loading: false,
        logout: async (...args) => console.log('logout', ...args),
        selectOrgId: (...args) => console.log('selectOrgId', ...args),
        redirectToLoginPage: (...args) =>
          console.log('redirectToLoginPage', ...args),
        redirectToSignupPage: (...args) =>
          console.log('redirectToSignupPage', ...args),
        redirectToAccountPage: (...args) =>
          console.log('redirectToAccountPage', ...args),
        redirectToOrgPage: (...args) =>
          console.log('redirectToOrgPage', ...args),
        redirectToCreateOrgPage: (...args) =>
          console.log('redirectToCreateOrgPage', ...args),
        userSelectedOrgId: 'demo.org.id',
      }}
    >
    <Component />
</AuthContext.Provider>
```

I can render a `Component` that relies on `@propelauth/react` in isolation without needing to rely on an actual auth endpoint. This is incredibly useful for unit testing these components and helps to remove the dependency on the auth endpoint when testing.

If there is a better way, please let me know.  Else, exposing the `AuthContext` seems like such a small but incredibly useful change.

**Comment by andrew-propelauth** [comment]
Hey, thanks for reaching out - we discussed this a bit separately, but our primary concern was that the AuthContext is essentially internal state. By exposing it, we would need to maintain backwards compatibility there, which is why we are hesitant about that PR.

We do think this is a very real use case though, so we made this PR: https://github.com/PropelAuth/react/pull/8 which creates an AuthProviderForTesting that acts similarly to the AuthProvider, but allows you to hardcode a user.

Does that work for this case?
